File: src/main/java/coffeetime/domain/type/PriceType.java
package coffeetime.domain.type;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum PriceType {
	CHEAP("가성비 있는"),
	AVERAGE("적당한"),
	EXPENSIVE("비싼");

	private final String price;

}

File: src/main/kotlin/coffeetime/enums/PriceType.kt
package coffeetime.enums

enum class PriceType(
	val value: String
) {
	CHEAP("가성비 있는"),
	AVERAGE("적당한"),
	EXPENSIVE("비싼");

	companion object {
		fun from(name: String): PriceType = entries.find { it.value == name }
			?: throw IllegalArgumentException("No enum constant for display name: $name")
	}
}
